src/test/kotlin/org/rust/lang/core/completion/RsCompletionTest.kt

```diff
@@ -675,6 +675,174 @@ class RsCompletionTest : RsCompletionTestBase() {
        }
     """)
 
+    @ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
+    fun `test complete paths in path constructor 1`() = doSingleCompletionByFileTree("""
+    //- main.rs
+        fn main() {
+            std::path::Path::new("fo/*caret*/");
+        }
+    //- foo.rs
+        pub struct Foo;
+    """, """
+        fn main() {
+            std::path::Path::new("foo.rs/*caret*/");
+        }
+    """)
+
+    @ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
+    fun `test complete paths in path constructor 2`() = doSingleCompletionByFileTree("""
+    //- main.rs
+        use std::path::Path;
+        fn main() {
+            Path::new("fo/*caret*/");
+        }
+    //- foo.rs
+        pub struct Foo;
+    """, """
+        use std::path::Path;
+        fn main() {
+            Path::new("foo.rs/*caret*/");
+        }
+    """)
+
+    /* TODO: enable when name resolution resolves <Path as Foo>::new to Foo::new
+    @ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
+    fun `test do not complete paths in path trait impl`() = checkNoCompletionByFileTree("""
+    //- main.rs
+        use std::path::Path;
+        trait Foo {
+            fn new(x: &str) -> i32;
+        }
+        impl Foo for Path {
+            fn new(x: &str) -> i32 {
+                123
+            }
+        }
+        fn main() {
+            <Path as Foo>::new("fo/*caret*/");
+        }
+    //- foo.rs
+        pub struct Foo;
+    """)*/
```

Comment:
We have `org.rust.RsTestBase#expect` method to write tests that should fail. Let's use it in this case. Otherwise, when someone fixes name resolution, this test will be still commented